src/main/java/org/takes/tk/TkRedirect.java

```diff
@@ -68,11 +70,59 @@ public TkRedirect(final String location, final int code) {
         super(
             new Take() {
                 @Override
-                public Response act(final Request req) {
-                    return new RsRedirect(location, code);
+                public Response act(final Request req) throws IOException {
+                    return new RsRedirect(
+                        new RedirectParams(req, location).location(),
+                        code
+                    );
                 }
             }
         );
     }
 
+    /**
+     * Extract params from original query.
+     */
+    private static final class RedirectParams {
```

Comment:
@neonailol I think you can use `RqHref(req).href()` instead of this new class

Reply:
@g4s8 if this is bug i can add pdd referencing this issue, for this class to be removed after it will be fixed

Reply:
@neonailol OK, I see

Reply:
@g4s8 i created an issue with details #793